nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -12,76 +12,207 @@
+  temp_tree.root_node = nullptr;
 
-  // Create and run the behavior tree
-  NavigateToPoseBehaviorTree bt(shared_from_this());
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+BtNavigator::on_activate(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(get_logger(), "Activating");
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+BtNavigator::on_deactivate(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(get_logger(), "Deactivating");
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
 
-  TaskStatus result = bt.run(blackboard, xml_string,
-      std::bind(&nav2_tasks::NavigateToPoseTaskServer::cancelRequested, task_server_.get()));
+nav2_lifecycle::CallbackReturn
+BtNavigator::on_cleanup(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(get_logger(), "Cleaning up");
+
+  goal_sub_.reset();
+  client_node_.reset();
+  self_client_.reset();
+  action_server_.reset();
+  path_.reset();
+  xml_string_.clear();
+  tree_.reset();
+  blackboard_.reset();
+  bt_.reset();
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+BtNavigator::on_error(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_ERROR(get_logger(), "Handling error state");
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+BtNavigator::on_shutdown(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(get_logger(), "Shutting down");
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+void
+BtNavigator::navigateToPose(const std::shared_ptr<GoalHandle> goal_handle)
+{
+  // Initialize the goal and result
+  auto goal = goal_handle->get_goal();
+  auto result = std::make_shared<nav2_msgs::action::NavigateToPose::Result>();
+
+  // TODO(mjeronimo): handle or reject an attempted pre-emption
+
+  RCLCPP_INFO(get_logger(), "Begin navigating from current location to (%.2f, %.2f)",
+    goal->pose.pose.position.x, goal->pose.pose.position.y);
+
+  // Get the goal pointer off of the blackboard...
+  auto bb_goal = blackboard_->get<geometry_msgs::msg::PoseStamped::SharedPtr>("goal");
+
+  // and update it with the incoming command
+  *bb_goal = goal->pose;
+
+  // Execute the BT that was previously created in the configure step
+  auto is_canceling = [goal_handle]() -> bool {return goal_handle->is_canceling();};
+  TaskStatus rc = bt_->run(tree_, is_canceling);
+
+  switch (rc) {
+    case TaskStatus::CANCELED:
+      // Even though the BT is no longer running, remote actions may still be executing. So,
+      // an explicit canceling of all actions allows the task clients to send cancel messages
+      // to their corresponding task servers
+      bt_->cancelAllActions(tree_->root_node);
+
+      // Reset the BT so that it can be run again in the future
+      bt_->resetTree(tree_->root_node);
+      goal_handle->set_canceled(result);
+      break;
+
+    case TaskStatus::SUCCEEDED:
+      goal_handle->set_succeeded(result);
```

Comment:
Action state transitions have been renamed in master branch of rclcpp_action, `goal_handle->set_succeeded` is now `goal_handle->succeed`

Reply:
Fixed.